### PR TITLE
Migrate to Java 17 switch 

### PR DESCRIPTION
**Pull request by preveen-stack**
Replace legacy switch block with java 17 semantics
ref: https://openjdk.org/jeps/361

PS: This is my first PR in eclipse-pde.

### CODE REVIEW

**Comment by vogella** [comment]
Thanks.

Please squash the commits into one and avoid merge commits in future contriburtions.

**Comment by preveen-stack** [comment]
ECA completed

**Review by HannesWell — COMMENTED** [review]
Thanks for this.
From a coarse look it looks good, I only have one little more suggestion below.
Have not yet reviewed in detail.

File: apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/search/UseScanParser.java [hunk 1]
```diff
@@ -142,28 +142,26 @@ protected void processElement(String uri, String localName, String name, Attribu
...
 				}
-				break;
-			case IApiXmlConstants.ELEMENT_TARGET:{
+			}
+			case IApiXmlConstants.ELEMENT_TARGET -> {
 				String qName = attributes.getValue(IApiXmlConstants.ATTR_TYPE);
 				String memberName = attributes.getValue(IApiXmlConstants.ATTR_MEMBER_NAME);
 				String signature = attributes.getValue(IApiXmlConstants.ATTR_SIGNATURE);
 				IMemberDescriptor member = null;
 				switch (type) {
```
[the remark this answers is not in the document]
Comment (preveen-stack) [reply apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/search/UseScanParser.java]: Made the suggested changes

**Comment by vogella** [comment]
LGTM

Any more suggestions @HannesWell ?

**Comment by HannesWell** [comment]
> LGTM
> 
> Any more suggestions @HannesWell ?

I'll review it tomorrow.

**Review by HannesWell — APPROVED** [review]
The change looks good.
But at some places I think the use of switches is not so well suited and if/else/else-if statements are a better fit. But that was already the case before.

Nevertheless I changed those places to use if-statements instead.
In general I think many of the switch-expressions should be pattern-matches, but that will require Java-21 or even later (see https://github.com/eclipse-pde/eclipse.pde/issues/592).

Besides that I added missing default cases that caused warnings.

**Comment by HannesWell** [comment]
Thank you @preveen-stack.

**Pull request merged**